quickwit/quickwit-doc-mapper/src/default_doc_mapper/mapping_tree.rs

```diff
@@ -246,20 +451,18 @@ fn insert_json_val(
 trait NumVal: Sized + FromStr + ToString + Into<TantivyValue> {
     fn from_json_number(num: &serde_json::Number) -> Option<Self>;
 
-    fn from_json(json_val: JsonValue, coerce: bool) -> Result<TantivyValue, String> {
+    fn from_json_to_self(json_val: JsonValue, coerce: bool) -> Result<Self, String> {
```

Comment:
```suggestion
    fn from_json(json_val: JsonValue, coerce: bool) -> Result<Self, String> {
```
I think to_self is implied?